test/metadata_closure_test.js

```diff
@@ -4,15 +4,15 @@ describe('A test', {
   metadata: { a: 1 },
 
   tests: function() {
-    it('should be able to access its own metadata, given by the suite', function(_metadata) {
-      expect(_metadata).to.eql({ a: 1 });
+    it('should be able to access its own metadata, given by the suite', function() {
+      expect(this.test.metadata).to.eql({ a: 1 });
```

Comment:
@mbroadst you were right! Here `this` refers to a [Context](https://github.com/mochajs/mocha/blob/master/lib/context.js) object, passed to each and test at runtime